### PR TITLE
[Feature][Connector-V2][File] Support skip number when reading text csv files

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #3276 #3192 

Because I have no permit to push his branch, so I open another pull request.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/FtpFile.md [hunk 1]
```diff
@@ -29,21 +29,22 @@ If you use SeaTunnel Engine, It automatically integrated the hadoop jar when you
...
+| user                      | string  | yes      | -                   |
+| password                  | string  | yes      | -                   |
+| path                      | string  | yes      | -                   |
+| type                      | string  | yes      | -                   |
+| delimiter                 | string  | no       | \001                |
+| parse_partition_from_path | boolean | no       | true                |
+| date_format               | string  | no       | yyyy-MM-dd          |
+| datetime_format           | string  | no       | yyyy-MM-dd HH:mm:ss |
+| time_format               | string  | no       | HH:mm:ss            |
+| skip_header_row_number    | long    | no       | 1                   |
```
Comment (TaoZex) [inline docs/en/connector-v2/source/FtpFile.md]:
```suggestion
| skip_header_row_number    | long    | no       | 0                   |
```

File: docs/en/connector-v2/source/LocalFile.md [hunk 2]
```diff
@@ -34,17 +34,18 @@ Read all the data in a split in a pollNext call. What splits are read will be sa
...
+| name                      | type    | required | default value       |
+|---------------------------|---------|----------|---------------------|
+| path                      | string  | yes      | -                   |
+| type                      | string  | yes      | -                   |
+| delimiter                 | string  | no       | \001                |
+| parse_partition_from_path | boolean | no       | true                |
+| date_format               | string  | no       | yyyy-MM-dd          |
+| datetime_format           | string  | no       | yyyy-MM-dd HH:mm:ss |
+| time_format               | string  | no       | HH:mm:ss            |
+| skip_header_row_number    | long    | no       | 1                   |
```
Comment (TaoZex) [inline docs/en/connector-v2/source/LocalFile.md]: Same as above.

File: docs/en/connector-v2/source/OssFile.md [hunk 3]
```diff
@@ -47,6 +47,7 @@ Read all the data in a split in a pollNext call. What splits are read will be sa
 | endpoint                  | string  | yes      | -                   |
 | delimiter                 | string  | no       | \001                |
 | parse_partition_from_path | boolean | no       | true                |
+| skip_header_row_number    | long    | no       | 1                   |
```
Comment (TaoZex) [inline docs/en/connector-v2/source/OssFile.md]: Same as above.

File: docs/en/connector-v2/source/SftpFile.md [hunk 4]
```diff
@@ -29,21 +29,22 @@ If you use SeaTunnel Engine, It automatically integrated the hadoop jar when you
...
+| host                      | string  | yes      | -                   |
+| port                      | int     | yes      | -                   |
+| user                      | string  | yes      | -                   |
+| password                  | string  | yes      | -                   |
+| path                      | string  | yes      | -                   |
+| type                      | string  | yes      | -                   |
+| delimiter                 | string  | no       | \001                |
+| parse_partition_from_path | boolean | no       | true                |
+| date_format               | string  | no       | yyyy-MM-dd          |
+| skip_header_row_number    | long    | no       | 1                   |
```
Comment (TaoZex) [inline docs/en/connector-v2/source/SftpFile.md]: Same as above.

**Review by TaoZex — DISMISSED** [review]
LGTM

**Review by Hisoka-X — DISMISSED** [review]

**Review by CalvinKirs — COMMENTED** [review]
LGTM, could you add a test case?

**Comment by TyrantLucifer** [comment]
> LGTM, could you add a test case?

Yup, it's in progress.

**Review by TaoZex — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**